code/__DEFINES/machines.dm

```diff
@@ -29,7 +29,8 @@
 #define BIOGENERATOR	(1<<5) 	//Uses biomass
 #define LIMBGROWER		(1<<6) 	//Uses synthetic flesh
 #define SMELTER			(1<<7) 	//uses various minerals
-#define NANITE_COMPILER  (1<<8) //Prints nanite disks
+#define NANITE_COMPILER (1<<8) //Prints nanite disks
+#define TANKFAB			(1<<9) //Snowflakes for hardpoints and their ammunitions.
```

Comment:
Horrible, horrible snowflaking. Just typecheck the data you get from the href and don't let the user set the cost.

Reply:
I tried doing that, but that was equally dumbstrucking to my sight and ever so slighty worse for future coding shenanigeans. Though this TANKFAB flag is mostly a dummy I admit, I can remove it if you want.

Reply:
Honestly just a typecache of things it's allowed to print, generate the cost in the machine instead of in the href, and you're set.